### PR TITLE
enabled remote getWorkspaceByDatabase

### LINKED ISSUE #750: [FEAT] - getWorkspaceByDatabase - Enable functionality for remote workspaces
`getWorkspaceByDatabase` is often used to get a workspace object but after the deployment of multi-workspace deployments we forgot to enable this utility to work for instantiating remote workspaces.

### PR DESCRIPTION
**Pull request by GeekSheikh**
closes #750

### CODE REVIEW

File: src/main/scala/com/databricks/labs/overwatch/utils/Tools.scala [hunk 1]
```diff
@@ -539,19 +539,29 @@ object Helpers extends SparkSessionWrapper {
...
+   * @param successfullOnly Only consider successful runs when looking for latest config
+   * @param disableValidations Whether or not to have initializer disable validations
    * @return
    */
-  def getWorkspaceByDatabase(etlDB: String, successfulOnly: Boolean = true, disableValidations: Boolean = false): Workspace = {
+  def getWorkspaceByDatabase(
+                              etlDB: String,
+                              organization_id: Option[String] = None,
+                              apiUrl: Option[String] = None,
+                              successfullOnly: Boolean = true,
```
Comment (sriram251-code) [inline src/main/scala/com/databricks/labs/overwatch/utils/Tools.scala:552]: successfullOnly what is the use of this field???

**Review by sriram251-code — CHANGES_REQUESTED** [review]

**Review by brij-raghuwanshi-db — CHANGES_REQUESTED** [review]
How or where are we getting the apiURL for remote workspaces?

**Review by sriram251-code — APPROVED** [review]

**Comment by GeekSheikh** [comment]
> How or where are we getting the apiURL for remote workspaces?

@brij-raghuwanshi-db here the user must look up the orgid and apiURL from their config to instantiate the remote workspace.

**Review by brij-raghuwanshi-db — APPROVED** [review]
Looks good.

**Pull request merged**